src/client.rs

```diff
@@ -156,42 +156,40 @@ where
     /// repositories using the provided key IDs to pin the verification.
     ///
     /// This is the preferred method of creating a client.
-    pub fn with_root_pinned<'a, I>(
-        trusted_root_keys: I,
+    pub async fn with_root_pinned(
+        trusted_root_keys: &[KeyId],
```

Comment:
Why did the type here change?

Reply:
I was getting the following error:

```rust
error[E0700]: hidden type for `impl Trait` captures lifetime that does not appear in bounds
   --> src/client.rs:168:10
    |
168 |     ) -> Result<Self>
    |          ^^^^^^^^^^^^
    |
note: hidden type `impl core::future::future::Future` captures the lifetime 'a as defined on the method body at 163:35
   --> src/client.rs:163:35
    |
163 |     pub async fn with_root_pinned<'a, I>(
    |                                   ^^
```

Wasn't sure if this was a limitation of impl Future/async/await, or it's actually supposed to work and it's a compiler error. Do you have a sense?

Reply:
This is supposed to work, but currently does not.